tests/odbc/test-odbc-mssql.cpp

```diff
@@ -22,127 +22,613 @@ backend_factory const &backEnd = *soci::factory_odbc();
+  {
+    CHECK(ch_out[i] == ch_in[i]);
+  }
+}
+
+// TODO: See if we can get this to work on Windows. The tests pass on Linux/MacOS.
+// It seems that on Linux the MS SQL ODBC driver does implicitly convert
+// between UTF-8 and UTF-16, but on Windows it doesn't.
+// For standard_into_type_backend it's possible to describe the column and
+// implicit conversion was therefore possible. But for the vector_into_type_backend
+// it that didn't work, as the call to describe_column() failed.
+
+// TEST_CASE("MS SQL string stream implicit unicode conversion", "[odbc][mssql][string][stream][utf8-utf16-conversion]")
+// {
+//     soci::session sql(backEnd, connectString);
+
+//     struct wide_text_table_creator : public table_creator_base
+//     {
+//         explicit wide_text_table_creator(soci::session& sql)
+//             : table_creator_base(sql)
+//         {
+//             sql << "create table soci_test ("
+//                 "wide_text nvarchar(40) null"
+//                 ")";
+//         }
+//     } wide_text_table_creator(sql);
+
+//     //std::string const str_in = u8"สวัสดี!";
+//     std::string const str_in = "\xe0\xb8\xaa\xe0\xb8\xa7\xe0\xb8\xb1\xe0\xb8\xaa\xe0\xb8\x94\xe0\xb8\xb5!";
+
+//     sql << "insert into soci_test(wide_text) values(N'" << str_in << "')";
+
+//     std::string str_out;
+//     sql << "select wide_text from soci_test", into(str_out);
+
+//     std::wstring wstr_out;
+//     sql << "select wide_text from soci_test", into(wstr_out);
+
+//     CHECK(str_out == str_in);
+
+// #if defined(SOCI_WCHAR_T_IS_WIDE) // Unices
+//     CHECK(wstr_out == L"\U00000E2A\U00000E27\U00000E31\U00000E2A\U00000E14\U00000E35\U00000021");
+// #else // Windows
+//     CHECK(wstr_out == L"\u0E2A\u0E27\u0E31\u0E2A\u0E14\u0E35\u0021");
+// #endif
+
+// }
+
+// TEST_CASE("MS SQL wide string stream implicit unicode conversion", "[odbc][mssql][wstring][stream][utf8-utf16-conversion]")
+// {
+//     soci::session sql(backEnd, connectString);
+
+//     struct wide_text_table_creator : public table_creator_base
+//     {
+//         explicit wide_text_table_creator(soci::session& sql)
+//             : table_creator_base(sql)
+//         {
+//             sql << "create table soci_test ("
+//                 "wide_text nvarchar(40) null"
+//                 ")";
+//         }
+//     } wide_text_table_creator(sql);
+
+//     //std::string const str_in = u8"สวัสดี!";
+//     std::wstring const wstr_in = L"\u0E2A\u0E27\u0E31\u0E2A\u0E14\u0E35\u0021";
+
+//     sql << "insert into soci_test(wide_text) values(N'" << wstr_in << "')";
+
+//     std::string str_out;
+//     sql << "select wide_text from soci_test", into(str_out);
+
+//     std::wstring wstr_out;
+//     sql << "select wide_text from soci_test", into(wstr_out);
+
+//     CHECK(str_out == "\xe0\xb8\xaa\xe0\xb8\xa7\xe0\xb8\xb1\xe0\xb8\xaa\xe0\xb8\x94\xe0\xb8\xb5!");
+//     CHECK(wstr_out == wstr_in);
+
+// }
+
+TEST_CASE("UTF-8 validation tests", "[unicode]")
```

Comment:
All these tests are neither MSSQL nor ODBC specific, they should ideally be in their own file.

Reply:
I have moved them to the "empty" test module, as it contains other non-backend-specific tests. 
I don't think this is the best solution, but I would need more information on how you want a separate unicode test file to be treated in the context of the CMake files. The backend tests use the CMake macro `soci_backend_test`.

I can just "nail it in", but I assume a more elegant solution is preferred.